R/dummy_multi_choice.R

```diff
@@ -0,0 +1,305 @@
+#'  Multi Choice Dummy Variables Creation
+#'
+#' `step_dummy_multi_choice()` creates a *specification* of a recipe
+#'  step that will convert multiple nominal data (e.g. character or factors)
+#'  into one or more numeric binary model terms for the levels of
+#'  the original data.
+#'
+#' @inheritParams step_center
+#' @inheritParams step_dummy
+#' @param ... One or more selector functions to choose which variables will be
+#'  used to compute the components. See [selections()] for more details.
+#' @param role For model terms created by this step, what analysis
+#'  role should they be assigned?. By default, the function assumes
+#'  that the binary dummy variable columns created by the original
+#'  variables will be used as predictors in a model.
+#' @param threshold A numeric value between 0 and 1. If less than one,
+#'  factor levels whose rate of occurrence in the training set are below
+#'  `threshold` will be "othered". See Details below.
+#'   If `threshold = 0` then no thresholding is done.
```

Comment:
Can this be the same as what is in `step_other()` to be consistent, and then inherited? At the least, we should use the same wording.